controllers/component_controller.go

```diff
@@ -310,13 +313,23 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 					devfileLocation = gitURL + string(os.PathSeparator) + devfileLocation
 				} else {
+					//cannot use converted URLs in SPI because it's not supported.  Need to convert later for parsing
+					gitURL = source.GitSource.URL
 					// Use SPI to retrieve the devfile from the private repository
-					devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, component.Namespace, source.GitSource.URL, source.GitSource.Revision, context)
+					devfileBytes, devfileLocation, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, component, gitURL, source.GitSource.Revision, context)
+					if err != nil {
+						log.Error(err, fmt.Sprintf("Unable to download from any known devfile locations from %s %v", gitURL, req.NamespacedName))
+						_ = r.SetCreateConditionAndUpdateCR(ctx, req, &component, err)
+						return ctrl.Result{}, err
```

Comment:
That was the existing logic I tried to preserve.  Doesn't the requeue happen if the error is not nil?  https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler.  I don't know why we don't handle the error for `_ = r.SetCreateConditionAndUpdateCR()` - maybe it's because we don't want to hide the real failure?